src/Core/Localization/CLDR/Currency.php

```diff
@@ -0,0 +1,183 @@
+     */
+    private $numericIsoCode;
+
+    /**
+     * Number of decimal digits to display for a price in this currency.
+     *
+     * @var int
+     */
+    private $decimalDigits;
+
+    /**
+     * Possible names depending on count context.
+     *
+     * e.g.: "Used currency is dollar" (default), "I need one dollar" (one), "I need five dollars" (other)
+     * [
+     *     'default' => 'dollar',
+     *     'one'     => 'dollar',
+     *     'other'   => 'dollars',
+     * ]
+     *
+     * @var string[]
+     */
+    private $displayNames;
+
+    /**
+     * Possible symbols (PrestaShop is using narrow).
+     *
+     * e.g.:
+     * [
+     *     'default' => 'US$',
+     *     'narrow' => '$',
+     * ]
+     *
+     * @var string[]
+     */
+    private $symbols;
+
+    public function __construct(CurrencyData $currencyData)
+    {
+        $this->isoCode = $currencyData->getIsoCode();
+        $this->numericIsoCode = $currencyData->getNumericIsoCode();
+        $this->decimalDigits = $currencyData->getDecimalDigits();
+        $this->displayNames = $currencyData->getDisplayNames();
+        $this->symbols = $currencyData->getSymbols();
+    }
+
+    /**
+     * Get the ISO code of this currency.
+     *
+     * @return string
+     *                The currency's ISO 4217 code
+     */
+    public function getIsoCode()
+    {
+        return $this->isoCode;
+    }
+
+    /**
+     * Get the numeric ISO code of this currency.
+     *
+     * @return string
+     *                The currency's ISO 4217 numeric code
+     */
+    public function getNumericIsoCode()
+    {
+        return $this->numericIsoCode;
+    }
+
+    /**
+     * Get the number of decimal digits to display when formatting a price with this currency.
+     *
+     * @return int
+     *             The number of decimal digits to display
+     */
+    public function getDecimalDigits()
+    {
+        return $this->decimalDigits;
+    }
+
+    /**
+     * Get the display name for the passed count context.
+     *
+     * @param string $countContext
+     *                             The count context
+     *                             "default" = talking about the currency (e.g.: "used currency is Euro")
+     *                             "one"     = talking about one unit of this currency (e.g.: "one euro")
+     *                             "other"   = talking about several units of this currency (e.g.: "ten euros")
+     *
+     * @return string
+     *                The wanted display name
+     */
+    public function getDisplayName($countContext = CurrencyInterface::DISPLAY_NAME_COUNT_DEFAULT)
+    {
+        if (!in_array($countContext, [CurrencyInterface::DISPLAY_NAME_COUNT_DEFAULT, CurrencyInterface::DISPLAY_NAME_COUNT_ONE, CurrencyInterface::DISPLAY_NAME_COUNT_OTHER])) {
+            throw new LocalizationException(
+                sprintf(
+                    'Unknown display name: "%s"',
+                    print_r($countContext, true)
```

Comment:
No need to print_r a string value.

Reply:
since we are not in php 7+ input is not locked as a string ;)

Reply:
well in this case prefer the annotation `(string) $countContext`

Reply:
what if given $countContext is an array ? ^^

Reply:
It will display `Array`,  and this is not our problem :)
We can not check all types, so cast it.

Reply:
Like you did here: https://github.com/PrestaShop/PrestaShop/pull/10455/files#diff-9da01ddf4abf062bf36197d7b0ce4e45R244